sinks/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/stripe/veneur/v14/samplers"
+	"github.com/stripe/veneur/v14/sinks"
+	"github.com/stripe/veneur/v14/ssf"
+	"github.com/stripe/veneur/v14/trace"
+	"github.com/stripe/veneur/v14/util"
+)
+
+const DefaultCloudwatchStandardUnitTagName = "cloudwatch_standard_unit"
+
+type CloudwatchMetricSinkConfig struct {
+	AWSRegion                     string             `yaml:"aws_region"`
+	CloudwatchEndpoint            string             `yaml:"cloudwatch_endpoint"`
+	CloudwatchNamespace           string             `yaml:"cloudwatch_namespace"`
+	CloudwatchStandardUnitTagName types.StandardUnit `yaml:"cloudwatch_standard_unit_tag_name"`
+}
+
+type cloudwatchMetricSink struct {
+	logger              *logrus.Entry
+	client              *cloudwatch.Client
+	endpoint            string
+	region              string
+	namespace           string
+	standardUnitTagName types.StandardUnit
+}
+
+func NewCloudwatchMetricSink(endpoint, namespace, region string, standardUnitTagName types.StandardUnit, logger *logrus.Entry) *cloudwatchMetricSink {
+	return &cloudwatchMetricSink{
+		endpoint:            endpoint,
+		namespace:           namespace,
+		region:              region,
+		standardUnitTagName: standardUnitTagName,
+		logger:              logger,
+	}
+}
+
+func ParseConfig(name string, config interface{}) (veneur.MetricSinkConfig, error) {
+	cloudwatchConfig := CloudwatchMetricSinkConfig{}
+	err := util.DecodeConfig(name, config, &cloudwatchConfig)
+	if err != nil {
+		return nil, err
+	}
+	if cloudwatchConfig.CloudwatchStandardUnitTagName == "" {
+		cloudwatchConfig.CloudwatchStandardUnitTagName = DefaultCloudwatchStandardUnitTagName
+	}
+	return cloudwatchConfig, nil
+}
+
+func Create(
+	server *veneur.Server, name string, logger *logrus.Entry,
+	config veneur.Config, sinkConfig veneur.MetricSinkConfig,
+) (sinks.MetricSink, error) {
+	conf, ok := sinkConfig.(CloudwatchMetricSinkConfig)
+	if !ok {
+		return nil, errors.New("invalid sink config type")
+	}
+
+	return NewCloudwatchMetricSink(
+		conf.CloudwatchEndpoint,
+		conf.CloudwatchNamespace,
+		conf.AWSRegion,
+		conf.CloudwatchStandardUnitTagName,
+		logger,
+	), nil
+}
+
+func (s *cloudwatchMetricSink) Name() string {
+	return "cloudwatch"
+}
+
+func (s *cloudwatchMetricSink) Start(*trace.Client) error {
+	opts := cloudwatch.Options{}
+	if s.endpoint != "" {
+		opts.EndpointResolver = cloudwatch.EndpointResolverFromURL(s.endpoint)
+	}
+	if s.region != "" {
+		opts.Region = s.region
+	}
+	s.client = cloudwatch.New(opts)
+	return nil
+}
+
+func (s *cloudwatchMetricSink) Flush(ctx context.Context, metrics []samplers.InterMetric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	metricData := make([]types.MetricDatum, len(metrics))
+	for i, metric := range metrics {
+		dimensions := make([]types.Dimension, 0, len(metric.Tags))
+		standardUnit := types.StandardUnitNone
+		for _, tag := range metric.Tags {
+			kv := strings.SplitN(tag, ":", 2)
+			if len(kv) < 2 {
+				continue // drop illegal tag
```

Comment:
Maybe worth calling out in the docs for this sink that it only supports tags in key:value format?

Reply:
This is an assumption baked into a lot of Veneur, I'd rather us tackle this separately.

Reply:
Note that I'd rather us not make this assumption, than for us to update the docs in the sinks that make this assumption.